model/convert/service_event.go

```diff
@@ -217,6 +231,7 @@ func convertServiceEventEpochSetup(event flow.Event) (*flow.ServiceEvent, error)
 			err,
 		)
 	}
+	setup.TargetEndTime = time.Unix(int64(targetEndTimeUnix), 0)
```

Comment:
> Lets assume we have nodes A and B that are located in different time zones, i.e. their returned Time object might have a different value in the internal field time.loc. Will EpochSetup.ID() return the same value for both nodes, despite Time being different for both nodes?

This is a good point - I don't think it will.

I'm less worried about the conversion to a `Time` type from unix time and back creating inconsistencies, but I agree avoiding it altogether is safer. I've modified the internal representation to use `uint64`.